src/cmd/tools/read_data_files/main/main.go

```diff
@@ -150,20 +150,26 @@ func main() {
 			start          = time.Now()
 		)
 
+		volumeNum, err := getVolumeNumber(*optPathPrefix, *optNamespace, int(shard), strconv.FormatInt(*optBlockstart, 10))
+		if err != nil {
+			log.Errorf("Failed to get volume number from file names for shard %d, using default shard number 0", shard)
+			volumeNum = int(*volume)
```

Comment:
The behavior doesn't match the message. It's to use the value from the command line flag instead of default value 0.

Reply:
fixed